clash-prelude/src/Clash/Explicit/BlockRam.hs

```diff
@@ -1241,99 +1249,264 @@ trueDualPortBlockRamWrapper clkA enA weA addrA datA clkB enB weB addrB datB =
+                          mem(~IF~SIZE[~TYP[#{addrB}]]~THENto_integer(~ARG[#{addrB}])~ELSE0~FI) := ~ARG[#{datB}];
+                      end if;
+                      ~SYM[#{symDoutB}] <= mem(~IF~SIZE[~TYP[#{addrB}]]~THENto_integer(~ARG[#{addrB}])~ELSE0~FI);
+                  end if;
+              end if;
+          end process;
+
+          ~RESULT <= (~SYM[#{symDoutA}], ~SYM[#{symDoutB}]);
+        end block;
+        -- end trueDualPortBlockRam
+|]) #-}
+{-# ANN trueDualPortBlockRam# (
+  let
+    bbName = show 'trueDualPortBlockRam#
+    (   _hasCallStack
+     :> knownNatAddrs
+     :> knownDomainA
+     :> knownDomainB
+     :> _nfdataX
+
+     :> clockA
+     :> enaA
+     :> wenaA
+     :> addrA
+     :> datA
+
+     :> clockB
+     :> enaB
+     :> wenaB
+     :> addrB
+     :> datB
+
+     :> Nil
+     ) = CV.indicesI @15
+
+    (   symMemName
+     :> symDoutA
+     :> symDoutB
+     :> Nil
+     ) = CV.indicesI @3
+  in
+    InlineYamlPrimitive [SystemVerilog] [__i|
+    BlackBox:
+      name: "#{bbName}"
+      kind: Declaration
+      template: |-
+        // trueDualPortBlockRam begin
+        // Shared memory
+        logic [~SIZE[~TYP[#{datA}]]-1:0] ~GENSYM[mem][#{symMemName}] [~LIT[#{knownNatAddrs}]-1:0];
+
+        ~SIGD[~GENSYM[data_slow][#{symDoutA}]][#{datA}];
+        ~SIGD[~GENSYM[data_fast][#{symDoutB}]][#{datB}];
+
+        // Port A
+        always @(~IF~ACTIVEEDGE[Rising][#{knownDomainA}]~THENposedge~ELSEnegedge~FI ~ARG[#{clockA}]) begin
+            if(~ARG[#{enaA}]) begin
+                ~SYM[#{symDoutA}] <= ~SYM[#{symMemName}][~IF~SIZE[~TYP[#{addrA}]]~THEN~ARG[#{addrA}]~ELSE0~FI];
+                if(~ARG[#{wenaA}]) begin
+                    ~SYM[#{symDoutA}] <= ~ARG[#{datA}];
+                    ~SYM[#{symMemName}][~IF~SIZE[~TYP[#{addrA}]]~THEN~ARG[#{addrA}]~ELSE0~FI] <= ~ARG[#{datA}];
+                end
+            end
+        end
+
+        // Port B
+        always @(~IF~ACTIVEEDGE[Rising][#{knownDomainB}]~THENposedge~ELSEnegedge~FI ~ARG[#{clockB}]) begin
+            if(~ARG[#{enaB}]) begin
+                ~SYM[#{symDoutB}] <= ~SYM[#{symMemName}][~IF~SIZE[~TYP[#{addrB}]]~THEN~ARG[#{addrB}]~ELSE0~FI];
+                if(~ARG[#{wenaB}]) begin
+                    ~SYM[#{symDoutB}] <= ~ARG[#{datB}];
+                    ~SYM[#{symMemName}][~IF~SIZE[~TYP[#{addrB}]]~THEN~ARG[#{addrB}]~ELSE0~FI] <= ~ARG[#{datB}];
+                end
+            end
+        end
+
+        assign ~RESULT = {~SYM[#{symDoutA}], ~SYM[#{symDoutB}]};
+        // end trueDualPortBlockRam
+|]) #-}
+{-# ANN trueDualPortBlockRam# (
+  let
+    bbName = show 'trueDualPortBlockRam#
+    (   _hasCallStack
+     :> knownNatAddrs
+     :> knownDomainA
+     :> knownDomainB
+     :> _nfdataX
+
+     :> clockA
+     :> enaA
+     :> wenaA
+     :> addrA
+     :> datA
+
+     :> clockB
+     :> enaB
+     :> wenaB
+     :> addrB
+     :> datB
+
+     :> Nil
+     ) = CV.indicesI @15
+
+    (   symMemName
```

Comment:
I feel the `Name` is superfluous; you don't use it for the other two symbols either. This is both SystemVerilog and Verilog.